go/vt/vtgate/executor.go

```diff
@@ -940,6 +943,46 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 			Rows:         rows,
 			RowsAffected: uint64(len(rows)),
 		}, nil
+	case "vschema columns":
+		if !show.HasOnTable() {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "show vschema column must provide table")
+		}
+
+		vschema := e.vm.GetCurrentSrvVschema()
+		if vschema == nil {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "vschema not loaded")
+		}
+
+		rows := make([][]sqltypes.Value, 0, 16)
+		// If the table reference is not fully qualified, then
+		// pull the keyspace from the session. Fail if the keyspace
+		// isn't specified or isn't valid, or if the table isn't
+		// known.
+		ksName := show.OnTable.Qualifier.String()
+		if ksName == "" {
+			ksName = destKeyspace
+		}
+
+		ks, ok := vschema.Keyspaces[ksName]
+		if !ok {
+			return nil, errNoKeyspace
+		}
+
+		tableName := show.OnTable.Name.String()
+		table, ok := ks.Tables[tableName]
+		if !ok {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "table `%s` does not exist in keyspace `%s`", tableName, ksName)
+		}
+
+		for _, col := range table.Columns {
+			rows = append(rows, buildVarCharRow(col.Name, col.Type.String()))
+		}
+
+		return &sqltypes.Result{
+			Fields:       buildVarCharFields("Name", "Type"),
+			Rows:         rows,
+			RowsAffected: uint64(len(rows)),
```

Comment:
The current behavior mimics the behavior of the mysql clients, which has two modes: `use_results` and `store_results`. One is streaming and the other is all-at-once. The all-at-once case does set `RowsAffected` even for selects, but the all-at-once case doesn't. Since this is a client side behavior (which doesn't parse queries), it's likely that it doesn't differentiate between select, show, etc.

Right now, I think this is relevant only for grpc. There's probably no way to send this info when returning query results.